pkg/reconciler/v1alpha1/service/resources/route.go

```diff
@@ -44,21 +47,34 @@ func MakeRoute(service *v1alpha1.Service) (*v1alpha1.Route, error) {
 		numRevisions := len(service.Spec.Release.Revisions)
 
 		// Configure the 'current' route.
-		currentRevisionName := service.Spec.Release.Revisions[0]
 		ttCurrent := v1alpha1.TrafficTarget{
-			Name:         "current",
-			Percent:      100 - rolloutPercent,
-			RevisionName: currentRevisionName,
+			Name:    "current",
+			Percent: 100 - rolloutPercent,
+		}
+		currentRevisionName := service.Spec.Release.Revisions[0]
+
+		// If the `current` revision refers to the well known name of the last
+		// known revision, use `Configuration` instead.
+		// Same for the `candidate` below.
+		// Part of #2819.
+		if currentRevisionName == releaseLastRevisionKeyword {
+			ttCurrent.ConfigurationName = names.Configuration(service)
```

Comment:
The trick for rollout completeness just works :D

Reply:
Totes  :-)